packages/utils/src/promisebuffer.ts

```diff
@@ -106,10 +102,19 @@ export class PromiseBuffer<T> {
       }, timeout);
 
       // if all promises resolve in time, cancel the timer and resolve to `true`
-      void allPromises(this._buffer).then(() => {
+      void allPromises(buffer).then(() => {
         clearTimeout(capturedSetTimeout);
         resolve(true);
       });
     });
   }
+
+  const promiseBuffer: PromiseBuffer<T> = {
+    _buffer: buffer,
```

Comment:
do we have to expose the buffer as well?

Reply:
@AbhiPrasad nope, just dropped it. The buffer is stored as _buffer on `BaseTransport` which was misleading when I searched